traj/src/traj/fanuc_limits.py

```diff
@@ -0,0 +1,134 @@
+    in the provided threshold tables (see below).
+
+    The payload argument is optional and will default to 'payload_max', as
+    provided in the call to 'gen_limit_interpolation_func(..)' (ie: this
+    function).
+
+    This will result in the slowest limit being returned, which would result
+    in a conservative estimate of the capabilities of the robot for the given
+    Cartesian velocity (but should not result in motion execution errors due
+    to violating a joint limit).
+
+    Depending on whether threshold tables for velocity, acceleration or jerk
+    are passed in, the limits returned are velocity, acceleration or jerk
+    limits.
+
+    The threshold tables are expected to conform to the format as returned by
+    a controller with J519 upon receipt of a 'Type 3' packet (ie: Request/Ack).
+    Assumptions are: 20 elements per table, elements of type float, sorted in
+    descending order (ie: max limit -> fully scaled down limit).
+
+    Args:
+        no_load_thresh: threshold table for NO load configuration
+          list(float)
+        max_load_thresh: threshold table for MAX load configuration
+          list(float)
+        payload_max: maximum payload supported by the robot (Kg)
+          float
+        cart_vmax: maximum Cartesian velocity supported by the robot (mm/s)
+          default: 4e3
+          float
+        num_steps: number of entries in a single threshold table
+          default: 20
+          int
+        interp_func: order of interpolation used. Passed on to interp2d(..)
+          default: 'linear'
+          str
+
+    Returns:
+        Function wrapping the return value of scipy.interp2d(..).
+
+        Args:
+          cart_vel: the Cartesian velocity of the TCP (mm/s)
+            float
+          payload: the weight of the current payload of the robot (Kg)
+            default: payload_max
+            float
+
+        Returns:
+          2D interpolated joint limit for the given Cartesian velocity and
+          payload.
+
+    Example:
+
+      # create interpolation function for the acceleration limits of J1, with
+      # a maximum payload of 25 Kg, and the default maximum Cartesian velocity
+      # (of 4000 mm/s), default number of elements in the threshold tables (20)
+      # and the default interpolation strategy (linear).
+      j1_acc_limit_func = gen_limit_interpolation_func(
+          no_load_thresh=[2050.00, 2050.00, ..],
+          max_load_thresh=[1601.56, 1601.56, ..],
+          payload_max=25.0
+      )
+
+      # determine acceleration limit for J1 with TCP moving at 1.5 m/s and
+      # with a current payload of 6.3 Kg
+      j1_curr_acc_limit = j1_acc_limit_func(cart_vel=1.5e3, payload=6.3)[0]
+
+      # determine acceleration limits for J1 with TCP moving at 1100, 1350 and
+      # 1470 mm/s and the default (ie: max) payload
+      j1_acc_limits = j1_acc_limit_func(cart_vel=[1100, 1350, 1470])
+    """
+    len_nlt = len(no_load_thresh)
+    len_mlt = len(max_load_thresh)
```

Comment:
Do you need to pass in num_steps? Couldn't you have:
```
num_steps = len(no_load_thresh)
if len(max_load_thresh) != num_steps:
    raise ValueError(...)
```

Reply:
Defensive programming I guess. By default all threshold tables have 20 elements right now. Having this argument here makes using different lengths tables more of a concious step.

Reply:
Makes sense.